Make new 7 interactive (#11442)
dotnet/docs: docs/csharp/whats-new/csharp-7.md

must still satisfy the async pattern, meaning a `GetAwaiter` method
must be accessible. As one concrete example, the `ValueTask` type
has been added to the .NET framework to make use of this new language
feature: 

[!code-csharp[UsingValueTask](~/samples/snippets/csharp/new-in-7/AsyncWork.cs#UsingValueTask "Using ValueTask")]

> [!NOTE]
> You need to add the NuGet package [`System.Threading.Tasks.Extensions`](https://www.nuget.org/packages/System.Threading.Tasks.Extensions/)
> in order to use the <xref:System.Threading.Tasks.ValueTask%601> type.

This enhancement is most useful for library authors to avoid allocating a `Task` in performance critical code.

## Numeric literal syntax improvements

Misreading numeric constants can make it harder to understand
code when reading it for the first time. Bit masks or other symbolic
values are prone to misunderstanding. C# 7.0 includes two new features to write numbers in the most readable fashion
for the intended use: *binary literals*, and *digit separators*.

For those times when you're creating bit masks, or whenever a
binary representation of a number makes the most readable code,
write that number in binary:

> [!code-csharp[ThousandSeparators](~/samples/snippets/csharp/new-in-7/Program.cs#33_ThousandSeparators "Thousands separators")]

Reviewer: ```suggestion
[!code-csharp[ThousandSeparators](~/samples/snippets/csharp/new-in-7/Program.cs#ThousandSeparators "Thousands separators")]
```

The `0b` at the beginning of the constant indicates that the
number is written as a binary number. Binary numbers can get long, so it's often easier to see
the bit patterns by introducing the `_` as a digit separator, as shown above in the binary constant. The digit separator can appear anywhere in the constant. For base 10

> numbers, it would be common to use it as a thousands separator:

Reviewer: ```suggestion
numbers, it is common to use it as a thousands separator:
```
